File: app/src/main/java/com/example/beto/viagemplanejada/Publicacao.java
package com.example.beto.viagemplanejada;

import java.util.Date;

public class Publicacao {
    private String pais;
    private String cidade;
    private Date dtViagem;
    private Date dtPublicacao;
    private float rating;

    public Publicacao(){}


    public Publicacao(String pais, String cidade, Date dtViagem, Date dtPublicacao, float rating) {
        this.pais = pais;
        this.cidade = cidade;
        this.dtViagem = dtViagem;
        this.dtPublicacao = dtPublicacao;
        this.rating = rating;
    }

    public String getPais() {
        return pais;
    }

    public void setPais(String pais) {
        this.pais = pais;
    }

    public String getCidade() {
        return cidade;
    }

    public void setCidade(String cidade) {
        this.cidade = cidade;
    }

    public Date getDtViagem() {
        return dtViagem;
    }

    public void setDtViagem(Date dtViagem) {
        this.dtViagem = dtViagem;
    }

    public Date getDtPublicacao() {
        return dtPublicacao;
    }

    public void setDtPublicacao(Date dtPublicacao) {
        this.dtPublicacao = dtPublicacao;
    }

    public float getRating() {
        return rating;
    }

    public void setRating(float rating) {
        this.rating = rating;
    }
}

File: app/src/main/java/com/example/beto/viagemplanejada/model/Publicacao.kt
package com.example.beto.viagemplanejada.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
@Entity
class Publicacao {
    var pais: String? = null
    var cidade: String? = null
    var dtViagem: Date? = null
    var dtPublicacao: Date? = null
    var rating: Float = 0.toFloat()
    @PrimaryKey var id:String=""


    constructor() {}


    constructor(pais: String, cidade: String, dtViagem: Date, dtPublicacao: Date, rating: Float) {
        this.pais = pais
        this.cidade = cidade
        this.dtViagem = dtViagem
        this.dtPublicacao = dtPublicacao
        this.rating = rating
    }
}
